File: Backend/ECORUTAAS/src/main/java/com/cibertec/models/Vehiculo.java
package com.cibertec.models;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnore;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "vehiculo")
public class Vehiculo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id_vehiculo;

    @Enumerated(EnumType.STRING)
    private Tipo tipo;

    private String codigo_qr;
    private Boolean disponible = true;
    private String ubicacion_actual;
    private LocalDateTime fecha_registro = LocalDateTime.now();

    @JsonIgnore
    @OneToMany(mappedBy = "vehiculo")
    private List<Alquiler> alquileres;

    public enum Tipo { 
        BICICLETA, 
        SCOOTER, 
        CAMINATA,    
        CARPOOL     
    }

}

File: FrontEnd/EcoRutas/src/models/vehiculo.ts
import { Alquiler } from './alquiler';

export interface Vehiculo {
  id_vehiculo?: number;
  tipo: 'BICICLETA' | 'SCOOTER' | 'MONOPATIN_ELECTRICO' | 'SEGWAY' | 'CARPOOL';
  codigo_qr: string;
  disponible: boolean;
  ubicacion_actual: string;
  fecha_registro?: string;
  alquileres?: Alquiler[];
}
